### PR TITLE
DialogServiceConnector, Connect/Disconnect impl

### PR DESCRIPTION
**Pull request by modnarb**
Implementations of the DialogServiceConnector and supporting classes, preliminary tests included.
Made changes to pass headers through the websocket initialization (was causing failures for service connection).
Removed code coverage parameter from jest config (causes broken source maps when in debugger)
Activity message response with TTS audio stream support.
DialogConfig split into BotFramework and CustomCommands configs.

### CODE REVIEW

**Comment by rhurey** [comment]
>         }

Do we still need to put them on the query line if the header's are being sent as headers now?

---
Refers to: src/common.browser/WebsocketConnection.ts:69 in b6c693e. [](commit_id = b6c693ee2b6563fd4cb9e8af2c5d426f520880ee, deletion_comment = False)

**Comment by rhurey** [comment]
>                 this.privWebsocketClient = new WebSocket(this.privUri);

Browser case needs the headers also.

---
Refers to: src/common.browser/WebsocketMessageAdapter.ts:105 in b6c693e. [](commit_id = b6c693ee2b6563fd4cb9e8af2c5d426f520880ee, deletion_comment = False)

File: tests/DialogServiceConnectorTests.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    const audioConfig: sdk.AudioConfig = sdk.AudioConfig.fromWavFileInput(file);
+
+    const connector: sdk.DialogServiceConnector = new sdk.DialogServiceConnector(connectorConfig, audioConfig);
+    objsToClose.push(connector);
+
+    expect(connector).not.toBeUndefined();
+    expect(connector instanceof sdk.DialogServiceConnector);
+});
+
+describe.each([true])("Service-based tests", (forceNodeWebSocket: boolean) => {
```
Comment (rhurey) [inline tests/DialogServiceConnectorTests.ts]: >true [](start = 15, length = 4)

true, false

**Comment by modnarb** [comment]
> > ```
> >     }
> > ```
> 
> Do we still need to put them on the query line if the header's are being sent as headers now?
> 
> Refers to: src/common.browser/WebsocketConnection.ts:69 in [b6c693e](https://github.com/microsoft/cognitive-services-speech-sdk-js/commit/b6c693ee2b6563fd4cb9e8af2c5d426f520880ee). [](commit_id = [b6c693e](https://github.com/microsoft/cognitive-services-speech-sdk-js/commit/b6c693ee2b6563fd4cb9e8af2c5d426f520880ee), deletion_comment = False)

Removing this would break the others, I didn't want bring that into this change.

File: tests/DialogServiceConnectorTests.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    expect(() => sdk.DialogServiceConfig.fromBotSecret(null, Settings.SpeechSubscriptionKey, null)).toThrowError();
+});
+
+test("Create DialogServiceConfig from bot secret, null Region", () => {
+    expect(() => sdk.DialogServiceConfig.fromBotSecret(Settings.BotSecret, Settings.SpeechSubscriptionKey, null)).toThrowError();
+});
+
+// test("Create DialogServiceConfig from bot secret, valid params", () => {
+
+// })
```
Comment (rhurey) [inline tests/DialogServiceConnectorTests.ts]: ?

File: src/common.speech/DialogConnectorFactory.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {
+    ProxyInfo,
+    WebsocketConnection,
+} from "../common.browser/Exports";
```
[the remark this answers is not in the document]
Comment (rhurey) [reply src/common.speech/DialogConnectorFactory.ts]: We should likely clean up the folder layout.

The original intent was something like:
common.browser: Things that deal with browser interaction. (The browser provides the web socket layer when not in node.)
common.speech: Things that are common for SR.
common: Things that are SR & browser agnostic, but common.

---
In reply to: [330116474](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/90#discussion_r330116474) [](ancestors = 330116474)

File: src/common.speech/DialogServiceAdapter.ts [hunk 4]
```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+import {
+    IAudioSource,
+    MessageType,
+    TranslationStatus,
```
[the remark this answers is not in the document]
Comment (modnarb) [reply src/common.speech/DialogServiceAdapter.ts]: Removing

File: src/sdk/ActivityReceivedEventArgs.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+/**
+ * Defines contents of speech recognizing/recognized event.
+ * @class SpeechRecognitionEventArgs
+ */
+export class ActivityReceivedEventArgs {
+    private privActivity: string;
```
[the remark this answers is not in the document]
Comment (modnarb) [reply src/sdk/ActivityReceivedEventArgs.ts]: Yes. Just FYI this class is getting a rename (in the coming week?) to something more generic like "message", e.g. MessageReceivedEventArgs

File: src/common.speech/DialogConnectorFactory.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            return botFramework;
+        }
+    }
+    throw new Error(`Invalid dialog type '${dialogType}'`);
+}
+
+export class DialogConnectionFactory implements IConnectionFactory {
+
+    public create = (
+        config: RecognizerConfig,
```
[the remark this answers is not in the document]
Comment (modnarb) [reply src/common.speech/DialogConnectorFactory.ts]: No, do you expect it there?

File: src/common.speech/DialogServiceAdapter.ts [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    // Cancels recognition.
+    protected cancelRecognition(
+        sessionId: string,
+        requestId: string,
+        cancellationReason: CancellationReason,
+        errorCode: CancellationErrorCode,
+        error: string,
+        cancelRecoCallback: (e: SpeechRecognitionResult) => void): void {
+        return;
+    //     if (!!this.privTranslationRecognizer.canceled) {
```
[the remark this answers is not in the document]
Comment (modnarb) [reply src/common.speech/DialogServiceAdapter.ts]: I'll remove altogether, the next PR will address this functionality more fully.

File: tests/DialogServiceConnectorTests.ts (hunk 2 above)
Comment (modnarb) [reply tests/DialogServiceConnectorTests.ts]: Missed that.. I'm adding more in the next PR, will address there

---
In reply to: [332231472](https://github.com/microsoft/cognitive-services-speech-sdk-js/pull/90#discussion_r332231472) [](ancestors = 332231472)

**Comment by modnarb** [comment]
> Changes in launch.json and jest.config.js are they desired? Looks good otherwise.

Reverted the code coverage in jest.config but leaving the launch.json as is.

**Pull request merged**